OpenEdXMobile/src/main/java/org/edx/mobile/view/adapters/CourseUnitPagerAdapter.java

```diff
@@ -60,15 +60,14 @@ public static boolean isCourseUnitVideo(CourseComponent unit) {
     public Fragment getItem(int pos) {
         CourseComponent unit = getUnit(pos);
         CourseUnitFragment unitFragment;
+        Boolean youtubeVideo = (unit instanceof VideoBlockModel && ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null);
         //FIXME - for the video, let's ignore studentViewMultiDevice for now
         if (isCourseUnitVideo(unit)) {
             unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit, (pos < unitList.size()), (pos > 0));
-        } else if (unit instanceof VideoBlockModel && ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
-            if (config.getYoutubeConfig().isYoutubeEnabled()) {
-                unitFragment = CourseUnitYoutubeVideoFragment.newInstance((VideoBlockModel) unit, (pos < unitList.size()), (pos > 0));
-            } else {
-                unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
-            }
+        } else if (youtubeVideo && config.getYoutubeConfig().isYoutubeEnabled()) {
```

Comment:
I like this. It looks really clean, but does this need to be changed to getEmbeddedYoutubeConfig to match the change in Config.java?

Reply:
I changed only the class name and kept the method name and this works but I am going to change the method name in order to keep consistency